test-engineering/contract-tests/client/tests/test_contile.py

```diff
@@ -3,34 +3,69 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 
-from typing import List
+from typing import Callable, Dict, List
 
 import pytest
 import requests
+from requests.models import Response
+
 from models import Step
 
 
-@pytest.fixture(name="contile_host")
-def fixture_contile_host(request):
-    """Read the contile host from the pytest config."""
+@pytest.fixture(name="hosts", scope="session")
+def fixture_hosts(request) -> Dict[str, str]:
+    """Return a dict mapping from a service name to a host name."""
+
+    return {
+        "contile": request.config.option.contile_url,
+        "partner": request.config.option.partner_url,
+    }
+
+
+@pytest.fixture(name="clear_partner_records")
+def fixture_clear_partner_records(hosts) -> Callable:
+    """Clear Contile request history on partner."""
+
+    partner_host = hosts["partner"]
+
+    def _clear_records():
+        r: Response = requests.request("DELETE", f"{partner_host}/records/")
+
+        error_message: str = (
+            f"Request status code was not as expected. Records may not have cleared "
+            f"after test run, details: '{r.text}'."
+        )
+
+        assert r.status_code == 204, error_message
```

Comment:
I'd probably raise an exception here instead of relying on assert to do that. Either way this should result in a pytest error rather than a failure because we clear records from a fixture, which is great!

Reply:
Change applied: [89450e7](https://github.com/mozilla-services/contile/pull/422/commits/89450e759fa97f84d6bd5f5e4ffc23060014084f)

I used the generic 'Exception'. Perhaps there is a specific one, along the lines of pytest.UsageError, that is more appropriate?

Reply:
There's only `UsageError` in the pytest API, so I'm afraid we need to write our own.

I typically define custom errors in a separate model, so they can be imported and caught by consumers of my code. In this case it's probably not necessary to create a separate module for that, but we could do something like this:

```python
class PartnerError(Exception):
    """Error specific to partner service interactions."""


class PartnerRecordsNotCleared(PartnerError):
    """Error clearning partner records."""

    def __init__(self, response: Response):
        error_message: str = (
            f"The Partner records may not have cleared after the test execution.\n"
            f"Response details:\n"
            f"Status Code: {response.status_code}\n"
            f"Content: '{response.text}'"
        )
        super().__init__(error_message)
```

and then 

```python
def clear_partner_records():
    response: Response = requests.delete(f"{partner_host}/records/")

    if response.status_code != 204:
        raise PartnerRecordsNotCleared(response)

return clear_partner_records
```

Reply:
Change applied: [529afcf](https://github.com/mozilla-services/contile/pull/422/commits/529afcf1b66575b0890b80c9498ab0aedcb6a112)